src/openzaak/js/components/admin/external-services/external-formset.js

```diff
@@ -0,0 +1,84 @@
+import React, {useState} from "react";
+import {ExternalForm} from "./external-form";
+import {ManagementForm} from "../../../forms/management-form";
+import PropTypes from "prop-types";
+
+const AddRow = ({ className="add-row", onAdd, children }) => {
+    return (
+        <tr className={className}>
+            <td colSpan="4">
+                <a href="#" onClick={onAdd}>{ children }</a>
+            </td>
+        </tr>
+    );
+};
+
+AddRow.propTypes = {
+    className: PropTypes.string,
+    onAdd: PropTypes.func.isRequired,
+};
+
+
+function ExternalFormSet(props) {
+    const { config, formData } = props;
+    const [ extra, setExtra ] = useState(config.TOTAL_FORMS - formData.length);
+
+    // set up the existing forms
+    const forms = formData.map(
+        (data, index) => <ExternalForm key={index} index={index} data={data} />
+    );
+
+     // build the extra forms in the formset based on the extra parameter
+    const numForms = forms.length;
+    const extraForms = Array(extra).fill().map(
+        (_, index) => <ExternalForm key={numForms + index} index={numForms + index} />
+    );
+
+    const allForms = forms.concat(extraForms);
+
+    return (
+        <div className="external-formset">
+            <ManagementForm
+                prefix={config.prefix}
+                initial_forms={config.INITIAL_FORMS}
+                total_forms={ formData.length + extra }
+                min_num_forms={config.MIN_NUM_FORMS}
+                max_num_forms={config.MAX_NUM_FORMS}
+            />
+
+            <fieldset className="module">
+                <table className="table">
+                    <caption>External services</caption>
+                    <thead>
+                        <tr>
+                            <th className='external-form__hidden'></th>
+                            <th className='external-formset__col'>Label</th>
+                            <th className='external-formset__col'>API type</th>
+                            <th className='external-formset__col--big'>Url</th>
+                            <th className='external-formset__col'>NLX</th>
+                            <th className='external-formset__col--big'>Documentation</th>
+                            <th className='external-formset__col'>Authorization</th>
+                            <th className='external-formset__col--small'>Verwijderen</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                    { allForms }
+
+                    <AddRow
+                        className="external-formset__add-row"
+                        onAdd={(event) => {
+                            event.preventDefault();
+                            setExtra(extra + 1);
+                        }}>
+                        Add Service
```

Comment:
Not 100% sure but prolly:

```suggestion
                        {% blocktrans with verbose_name=Service %}Add {{ verbose_name }}{% endblocktrans %}
```

See other Django add-buttons on exact format.

Reply:
afaik django templates are used when the page is refreshed, but react renders components without refreshing the page, so we can't use django template language inside react.

Reply:
I think you can but not worth it for now.

If interested, see: https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#internationalization-in-javascript-code

Reply:
Thanks! Didn't know about it

Reply:
just to add - internationalization in Django for JS is kinda... shoddy. Better to use something like `React.Intl` instead and handle it in the frontend stack.